### PR TITLE
PORTALS-1912: if image scale is set to 100, set to auto.  add katex preprocessing (…

### PR DESCRIPTION
**Pull request by jay-hodgson**
…same as Synapse.org).  Also, only process the math element in the DOM once.

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Here's the Hackthon wiki page after the changes (with fixed inline math processing, and showing the default image scale):
Screen Shot 2021-04-26 at 4 01 45 PM

File: src/lib/utils/SynapseClient.ts [hunk 1]
```diff
@@ -1114,6 +1114,11 @@ export const getPresignedUrlForWikiAttachment = (
     BackendDestinationEnum.REPO_ENDPOINT,
   )
 }
+
+export const isInSynapseExperimentalMode = ():boolean => {
```
Comment (jay-hodgson) [inline src/lib/utils/SynapseClient.ts:1118]: Also, add a way to ask if Synapse Experimental Mode is on (if the cookie exists).  Will use this to start to use the new Download List services.

File: styleguide.config.js [hunk 2]
```diff
@@ -102,15 +102,21 @@ module.exports = {
             {
                 src:"https://cdn.jsdelivr.net/npm/sanitize-html@1.20.0/dist/sanitize-html.min.js",
             },
+            {
+                src:"https://cdn.jsdelivr.net/npm/katex@0.10.0/dist/katex.min.js",
```
Comment (jay-hodgson) [inline styleguide.config.js:106]: Load katex in the StyleGuide

File: styleguide.config.js [hunk 3]
```diff
@@ -102,15 +102,21 @@ module.exports = {
...
         links: [
             {
               rel: 'stylesheet',
-              href:
-                'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css'
+              href: 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css'
+            },
+            {
+                rel: 'stylesheet',
+                href: 'https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css'
```
Comment (jay-hodgson) [inline styleguide.config.js:116]: And load the Katex css

File: src/lib/containers/MarkdownSynapse.tsx [hunk 4]
```diff
@@ -262,13 +262,18 @@ export default class MarkdownSynapse extends React.Component<
       '[id^="mathjax-"]',
     )
     // go through all obtained elements and transform them with katex
+    const regEx = new RegExp(/\\[()[\]]/, 'g') // Look for a '\' followed by either '(', ')', '[', or ']'. We delete these strings since they interfere with katex processing.
```
Comment (jay-hodgson) [inline src/lib/containers/MarkdownSynapse.tsx:265]: This syntax is a way to indicate that the math should be display inline, but we handle this outside of the latex (in the markdown it's either an inline math element or a block element).
https://www.overleaf.com/learn/latex/mathematical_expressions

File: src/lib/containers/MarkdownSynapse.tsx [hunk 5]
```diff
@@ -262,13 +262,18 @@ export default class MarkdownSynapse extends React.Component<
...
     mathExpressions.forEach(element => {
-      element.textContent &&
-        katex.render(element.textContent, element, {
+      if (element.textContent && !element.getAttribute('processed')) {
+        // only process a math element once, used to double/triple process
+        element.setAttribute('processed', 'true')
```
Comment (jay-hodgson) [inline src/lib/containers/MarkdownSynapse.tsx:269]: Only run the element content through katex processing once.

File: src/lib/containers/widgets/SynapseImage.tsx [hunk 6]
```diff
@@ -108,7 +108,12 @@ class SynapseImage extends React.Component<
   public render() {
     const { params } = this.props
     const { align = '', altText = 'synapse image' } = params
-    let scale = `${Number(params.scale) ?? 100}%`
+
+    let scale = 'auto'
+    if (params.scale && params.scale !== '100') {
+      scale = `${Number(params.scale)}%`
```
Comment (jay-hodgson) [inline src/lib/containers/widgets/SynapseImage.tsx:114]: Only use the scale parameter if it's set and not '100'.  Default value is 'auto'

File: src/lib/utils/SynapseClient.ts (hunk 1 above)
Comment (emmanmills) [reply src/lib/utils/SynapseClient.ts:1118]: This is a nice addon, we can use this for other feature testing too.

**Review by emmanmills — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**